### DISCUSSION TITLE
Shadercache and compatdata to SD card.

### DISCUSSION THREAD

**Discussion by tweakbsd** [mcarlucci/decky-storage-cleaner#5] [Ideas]
Hi there,

I love your Plug-In it is very well thought out and I like your implementation as well. The idea to move "compatdata" or "shadercache" folder to the SD card and symlink'ing them back on the SSD is not new but I think it would be a great feature for this plugin. I can write the Python code to do that and make a pull request if PRs are accepted. I have not much experience with React but adding another actions like the current "Clear selected" shouldn't be too hard. The action could be simply be named "Move selected compatdata to SD card" or similiar. 

I'd love to hear some thoughts on this topic. Currently I'm moving the big "compatdata" folders to the SD Card to save space on the SSD.

**Comment by mcarlucci** [comment]
Hi,

Thanks for the kind words and offer to contribute!

I think this feature makes sense and I have it on my radar/roadmap, but I haven't given it too much thought just yet. So I am glad you are placing some emphasis on it and offering to help. Much appreciated! 

I like your idea of adding a simple button and I think that could be a simple enough approach to start. However, here are some things to think about and contend with:

1.  We are working with limited horizontal real estate (slim slide out right now) for the most part. Being a traditional front end dev, I am passionate about creating a good (or at least, good _enough_) user experience. So adding more actions can be challenging as we can only add things vertically; to the top or bottom of the already potentially long (depends on number of games) scrollable slide out. This isn't the worst thing ever, just something to consider from users' perspectives - who are probably using their gamepad controls to navigate (as opposed to the touchscreen).
2. I think this feature and others could use some more prime real estate for a better experience, so I think adding a full screen page with tabs might be the way to go moving forward. The slide out area would become a menu for navigating to the appropriate tab in the full page layout and/or an area for quick action buttons. CSS Loader uses a full page layout for displaying different theming options, so I was thinking something similar to that.
3. In regards to moving the shader cache and compat data to the SD Card and symlinking feature, I think for now it could be fine to add a button to each section or group them together in their own section. One question I have is how would a user manage after moving to their SD card? Would the button have text and the functionality to move the data back to the SSD?

Please feel free to create a prototype/PR and I am more than happy to review. Also, please let me know if you have any questions or need any help.

**Comment by ADHDefy** [comment]
Is this idea sort of like the feature built into Cryo Utilites where you can hit a button and it will move things around so that the compat data and shader cache files are all on the same drives as their respective games? If so, I would _very_ much like to see this implemented. I no longer use Cryo Utilities, and it would be amazing to have that feature available in a Decky plugin like this.
